modules/transactions/src/integrationTest/java/org/apache/ignite/internal/disaster/DisasterRecoveryTestUtil.java

```diff
@@ -125,27 +126,34 @@ static void assertValueOnSpecificNodes(
             int id,
             int val,
             SchemaDescriptor schema
-    ) throws Exception {
-        for (IgniteImpl node : nodes) {
-            assertValueOnSpecificNode(tableName, node, id, val, schema);
-        }
+    ) {
+        CompletableFuture<?>[] futures = nodes.stream()
+                .map(node -> CompletableFuture.runAsync(() ->
+                        assertValueOnSpecificNode(tableName, node, id, val, schema)))
+                .toArray(CompletableFuture[]::new);
+
+        CompletableFuture.allOf(futures).join();
     }
 
-    static void assertValueOnSpecificNode(String tableName, IgniteImpl node, int id, int val, SchemaDescriptor schema) throws Exception {
+    static void assertValueOnSpecificNode(String tableName, IgniteImpl node, int id, int val, SchemaDescriptor schema) {
         InternalTable internalTable = unwrapTableViewInternal(node.tables().table(tableName)).internalTable();
 
         Row keyValueRow0 = createKeyValueRow(schema, id, val);
         Row keyRow0 = createKeyRow(schema, id);
 
-        assertTrue(waitForCondition(() -> {
-            try {
-                CompletableFuture<BinaryRow> getFut = internalTable.get(keyRow0, node.clock().now(), node.node());
-
-                return compareRows(getFut.get(), keyValueRow0);
-            } catch (Exception e) {
-                return false;
-            }
-        }, 10_000), "Row comparison failed within the timeout.");
+        try {
+            await()
+                    .atMost(ofSeconds(20))
+                    .until(() -> {
+                        BinaryRow actual = internalTable.get(keyRow0, node.clock().now(), node.node())
+                                .join();
+
+                        return compareRows(actual, keyValueRow0);
+                    });
+        } catch (ConditionTimeoutException e) {
+            throw new AssertionError("Row comparison failed within the timeout on " + node.name()
+                    + " for id=" + id + ", val=" + val, e);
```

Comment:
Should we obey `[id={}, val={}].` log messages notation in tests there? In utils class I think more as yes.

Reply:
done